pkg/render/node.go

```diff
@@ -1484,11 +1893,20 @@ func GetIPv6Pool(pools []operatorv1.IPPool) *operatorv1.IPPool {
 	return nil
 }
 
+// vppDataplaneEnabled returns true if the given Installation enables VPP
+func vppDataplaneEnabled(cr *operatorv1.InstallationSpec) bool {
+	return cr.CalicoNetwork != nil &&
+		cr.CalicoNetwork.LinuxDataplane != nil &&
+		*cr.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneVPP &&
+		cr.CalicoNetwork.VPP != nil
+}
+
 // bgpEnabled returns true if the given Installation enables BGP, false otherwise.
 func bgpEnabled(instance *operatorv1.InstallationSpec) bool {
 	return instance.CalicoNetwork != nil &&
 		instance.CalicoNetwork.BGP != nil &&
-		*instance.CalicoNetwork.BGP == operatorv1.BGPEnabled
+		*instance.CalicoNetwork.BGP == operatorv1.BGPEnabled &&
```

Comment:
I think this change is a bit confusing - instance.CalicoNetwork.BGP is meant to indicate whether or not BGP should be enabled - since VPP bundles its own BGP daemon, BGP is actually enabled, it's just not using BIRD.

Maybe it's as simple as renaming this function to `birdEnabled`? 

We probably also want to update validation to ensure that if VPP is enabled, instance.CalicoNetwork.BGP is also enabled - unless VPP data plane also supports running without BGP?